transport-native-epoll/src/main/c/netty_epoll_linuxsocket.c

```diff
@@ -256,7 +256,14 @@ static jint netty_epoll_linuxsocket_isTcpQuickAck(JNIEnv* env, jclass clazz, jin
 
 static jint netty_epoll_linuxsocket_isTcpFastOpenConnect(JNIEnv* env, jclass clazz, jint fd) {
     int optval;
-    if (netty_unix_socket_getOption(env, fd, IPPROTO_TCP, TCP_FASTOPEN_CONNECT, &optval, sizeof(optval)) == -1) {
+    int optlen = sizeof(optval);
+    // We call getsockopt directly so we can handle ENOPROTOOPT by ourself.
+    if (getsockopt(fd, IPPROTO_TCP, TCP_FASTOPEN_CONNECT, &optval, &optlen) == -1) {
```

Comment:
ideally we wouldn't have to call `getsockopt` directly and instead use a method from `netty_unix_socket` (e.g. expose `netty_unix_socket_getOption0` as something like `netty_unix_socket_getOptionNoErrorHandling`)

Reply:
If you pe der I can add it to the header file and call it... let me do a prq

Reply:
sgtm ... just lets us isolate the OS system calls more easily and can be useful when there are differences between platforms.

Reply:
@Scottmitch https://github.com/netty/netty/pull/7425 PTAL